src/Build/Evaluation/Expander.cs

```diff
@@ -2553,6 +2553,30 @@ internal static ItemTransformFunction GetItemTransformFunction(IElementLocation
                     }
                 }
 
+                /// <summary>
+                /// Intrinsic function that returns a boolean to indicate if any of the items have the given value
+                /// Using a case insensitive comparison
+                /// </summary>
+                internal static IEnumerable<Pair<string, S>> ContainsItem(Expander<P, I> expander, IElementLocation elementLocation, bool includeNullEntries, string functionName, IEnumerable<Pair<string, S>> itemsOfType, string[] arguments)
+                {
+                    ProjectErrorUtilities.VerifyThrowInvalidProject(arguments != null && arguments.Length == 1, elementLocation, "InvalidItemFunctionSyntax", functionName, (arguments == null ? 0 : arguments.Length));
+
+                    var hasItem = false;
+
+                    var searchedItem = arguments[0];
+
+                    foreach (var item in itemsOfType)
+                    {
+                        if (item.Key.Equals(searchedItem, StringComparison.OrdinalIgnoreCase))
+                        {
+                            hasItem = true;
+                            break;
+                        }
+                    }
+
+                    yield return new Pair<string, S>(hasItem ? "true" : "false", null /* no base item */);
```

Comment:
This is pretty different from the return pattern in `AnyHaveMetadataValue` which is otherwise similar:

https://github.com/Microsoft/msbuild/blob/4919c286d3bbf5cf1740860eefaea2b8c4e21b94/src/Build/Evaluation/Expander.cs#L2615-L2627

Do you know what changes downstream if we don't return the first matching item?